mlrun/api/db/sqldb/db.py

```diff
@@ -543,6 +620,8 @@ def list_artifacts(
         artifact_records = self._find_artifacts(
             session, project, ids, labels, since, until, name, kind, category, iter
         )
+        if as_records:
+            return artifact_records
```

Comment:
Putting this here means that you're not doing the processing of the `best_iteration` which comes after this point. This means that you'll not filter-out artifacts which do not belong to the best-iteration. 
Of course, this may be a valid assumption and we don't care about it, but then please add a validation that if `as_records` is `True` then `best_iteration` must be `False`.